controllers/favorites.js

```diff
@@ -3,40 +3,49 @@ module.exports = (app) => {
   const RecipeSchema = require('../models/recipe');
   const UserSchema = require('../models/user');
 
-  // Show all recipes saved in their cart
-  app.get('/favorites', (req, res) => {
-    // find recipe, return ones that have been favorited 1+ times
-    RecipeSchema.find({
-      'usersWhoFavorited': {
-        $ne: null
-      }
-    }, function(err, recipes) {
-      if (err) {
-        console.error(err);
-      } else {
-        res.render('favorites', {
-          recipes: recipes
-        });
-      }
-    })
-  });
-
   // Send a POST request to the database to create the recipes collection
   app.post('/favorites/', (req, res) => {
     const favoriteId = req.body.favoriteId;
     const userId = app.locals.user.id
 
     // find recipe, add userId to usersWhoFavorited
-    RecipeSchema.findByIdAndUpdate(favoriteId, {
-      $addToSet: { usersWhoFavorited: userId }}, function(err) {
+    RecipeSchema.findById(favoriteId, function(err, favoriteInDB) {
       if (err) return handleError(err);
+      if (favoriteInDB.usersWhoFavorited.includes(userId)) {
```

Comment:
suggestion for naming. it's good practice to indicate that it's an array of userIds instead of users. it's probably not something for this PR, perhaps a refactor PR, but perhaps something like `favoritedUserIds`